test/ejabberd_tests/tests/mam_SUITE.erl

```diff
@@ -425,166 +494,116 @@ do_init_per_group(C, ConfigIn) ->
             Config0
     end.
 
+end_per_group(rsm_all, Config) ->
+    Config;
+end_per_group(mam02, Config) ->
+    Config;
+end_per_group(mam03, Config) ->
+    Config;
+end_per_group(mam04, Config) ->
+    Config;
+end_per_group(rsm02, Config) ->
+    Config;
+end_per_group(rsm03, Config) ->
+    Config;
+end_per_group(rsm04, Config) ->
+    Config;
+end_per_group(with_rsm02, Config) ->
+    Config;
+end_per_group(with_rsm03, Config) ->
+    Config;
+end_per_group(with_rsm04, Config) ->
+    Config;
+end_per_group(mam_purge, Config) ->
+    Config;
+end_per_group(nostore, Config) ->
+    Config;
+end_per_group(archived, Config) ->
+    Config;
+end_per_group(mam_metrics, Config) ->
+    Config;
+end_per_group(muc02, Config) ->
+    Config;
+end_per_group(muc03, Config) ->
+    Config;
+end_per_group(muc04, Config) ->
+    Config;
+end_per_group(muc_rsm_all, Config) ->
```

Comment:
I'd move this one just before the generic the last end_per_group function head. It will improve readability and show which groups end by returning `Config` only.